### ISSUE TITLE
Save images to Bluesky folder

### ISSUE THREAD

**Issue by thktkblt** [bluesky-social/social-app#1360]
For now, the images downloaded from Bluesky are saved to general DCIM folder (I use Android), so it would get mixed with pics from other sources and feel cluttered.

**Comment by pfrazee** [comment]
I originally did this, but it required more permissions than folks were comfortable with :(

**Comment by fuelflo** [comment]
I'd also appreciate this. What permissions does that need, normally just access to storage and or photos/videos and you could only ask for permission when saving, right?

**Comment by pfrazee** [comment]
IIRC it forced me to ask for full photo library access (eat me). It's way out of scope for the team's current workstream but as a PR I'd accept some settings for configuring where your images are saved and dealing with the permissions.

**Comment by fuelflo** [comment]
Yeah, granular photo/video library access is only available from Android 14, which will probably be released in two days....

https://developer.android.com/about/versions/14/changes/partial-photo-video-access

But as Android manufacturer's upgrade slowly to newer versions, most of the users will be stuck for quite some time to have to grant full access.

**Comment by SaphireLattice** [comment]
Sigh. This one is getting extremely annoying. `/DCIM` folder gets very enthusiastically synced by Google Photos. Which uh, is extremely not what I want, and there's no way to exclude that without also excluding actual camera photos, I think?

So I end up with a mess of a gallery thanks to this.

**Comment by kairusds** [comment]
i wish they could just have the photos saved in Download/BlueSky or Picture/BlueSky

**Comment by dbrian57** [comment]
Came here to add my +1 for this change. The current setting really gunks up my camera roll. I like to keep my real life and weirdo online life separate, thank you very much.

**Comment by marksteward** [comment]
No extra permission should be needed to download files to the Downloads directory on Android.

**Comment by Nonplayable** [comment]
Begging y'all to let us save images downloaded from Bluesky to literally any folder other than DCIM. I do not need my camera folder or Google backups cluttered with dumb memes I want to send to a non Bluesky using friend

**Comment by RedFoxNightfox** [comment]
I too find it very annoying that I cannot change the download folder to not be DCIM. Every other andriod app I have ever used either allows me to change the download folder, or at least defaults to .\pictures\{app} or .\downloads\{app} folders.

**Comment by holistic-developer** [comment]
Since DCIM = digital camera images, this folder is not intended to be used for this purpose.

I understand the issue of asking too many permissions upfront, so ideally the required permissions could be requested right when the user tries to save an image.
This way users would see the direct connection between their action and the permissions necessary to fulfill them.

**Comment by SarcevicAntonio** [comment]
Y'all need to fix this, seriously makes the app useless to me and I'm stuck on the web version.

**Comment by tianon** [comment]
A simple workaround that I have been using successfully is that https://play.google.com/store/apps/details?id=com.google.android.apps.nbu.files has a share intent of "Download", so instead of clicking "Save" I click "Share" and choose "Files by Google | Download" and it puts the result into my downloads folder instead of into my camera roll (which isn't perfect and it's certainly an annoying flow, but for my uses it's a lot _less_ annoying than the files going straight into my camera roll as if I took them myself :shrug:).

**Comment by lordkitsuna** [comment]
This is desperately needed. I was so confused when suddenly a bunch of images from Blue Sky started showing up in my camera roll feed and that is absolutely not what I want I use blue sky to follow many artists and a lot of them draw inappropriate material and that's perfectly fine and I like saving that but I want it in the pictures folder or the downloads definitely not my main camera roll.

I don't know why anybody would be upset with the full image access permission that's completely normal on Android for an application like this to request when saving a file you will get asked the same thing if you use Discord Telegram and even many Matrix clients if you want something a little more open source. You'll get asked the same thing by many Lemmy clients and so on it's just a standard part of doing business on Android for now

I would imagine that there are many many more people upset about the fact that these images are ending up in their main camera feed rather than the app asking for gallery permission

**Comment by SaphireLattice** [comment]
It's getting pretty annoying to clear out stuff from Google Photos. 

And yeah, a permission asked when saving sounds all ok, and it's contextually understandable as to why the app's asking about it. 

Would be even better if we could select specific destination too...

**Comment by ticpu** [comment]
When I found out that Bluesky was actually saving furry art to my Google Photos I immediately removed access to Photos and medias to make sure I don't make this mistake ever again.

This is literally the worst folder to save media in without prior consent.  I see there's a pull request but I've been very surprised that Bluesky can request *write* access to the root of that folder at all.

**Comment by RESizemore** [comment]
Please add the option to change the download folder, it is super annoying everytime I choose to download an image to then have to immediately move it to a Bluesky folder that I created.

**Comment by julianfeder** [comment]
In my opinion this whole thing makes the Android App feel like an afterthought. No other app I know of behaves like this. Therefore I actually believe this is quite an important issue and one that meaningfully degrades the user experience.

**Comment by Ragerock** [comment]
It's almost like they don't care about the android users. Twitter is imploding and they could easily steal away so many more users with some quality of life updates such as this. The hesitation is confusing me.

**Comment by mozzius** [comment]
We're finally able to fix this. Thank you for your patience 😅
